mev-boost-rs/src/relay.rs

```diff
@@ -0,0 +1,78 @@
+use std::ops::Deref;
+
+use beacon_api_client::Client as BeaconClient;
+use ethereum_consensus::{
+    crypto::Error as CryptoError, primitives::BlsPublicKey, serde::try_bytes_from_hex_str,
+};
+use url::Url;
+
+use mev_rs::blinded_block_provider::Client;
+
+#[derive(Clone, Debug)]
+pub struct RelayEndpoint {
+    url: Url,
+    public_key: BlsPublicKey,
+}
+
+impl TryFrom<Url> for RelayEndpoint {
+    type Error = CryptoError;
+
+    fn try_from(url: Url) -> Result<Self, Self::Error> {
+        let public_key = try_bytes_from_hex_str(url.username())?;
+        let public_key = BlsPublicKey::try_from(&public_key[..])?;
```

Comment:
we can keep this `RelayEndpoint` type if you want but to be clear it is adding a lot of code just to wrap `try_bytes_from_hex_str` and `BlsPublicKey::try_from`

this could easily just be one function that lives where the url parsing code is currently defined

Reply:
yea, the other bit is that it replaces the [intermediate relay type](https://github.com/ralexstokes/mev-rs/blob/0332af04d0777625aea5e1be3608d4aef341d505/mev-boost-rs/src/service.rs#L51) where `Url` is. is there a reason that the `Relay` instances are not created until `spawn`?

Reply:
the reasoning is that I generally want to avoid `new` doing a bunch of effectful stuff -- `new` should just give you a thing and then, if you desire, you can start doing stuff w/ the thing